### ISSUE TITLE
Do not always load remote package info for path repository packages

### ISSUE THREAD

**Issue by sveneld** [composer/composer#10482]
In this issue https://github.com/composer/composer/issues/9751 updating of path repository was added, but
when many developers works with same package which is configured as 'path' they get a conflict during merging their branches due to the change of reference for path package in composer.lock. Is it possible to add an config option for path packages for removing such behavior?

**Comment by Seldaek** [comment]
IMO the downsides are worse than the conflicts you may get on the reference. But one question to me is whether we could also always fix the reference to a single value when it comes from a path repo, that would probably remove the conflicts entirely. I need to think some more about this but I think it works.

**Comment by Seldaek** [comment]
On second thought, the problem really is that if you don't change the reference every time you may have an out of date path package if your path repo is copying and not symlinking.

So I am not sure what we can do at least if copying is allowed. Perhaps we could suppress the git reference if symlink:true is set in the repository options.

**Comment by naderman** [comment]
Agreed: default/copying requires the reference, for anyone who wants to suppress the conflicts, we can force symlink:true and then leave the reference empty

**Comment by Seldaek** [comment]
OK abusing symlink:true is a bit of a mess after all IMO so I'd rather add a new option `reference` which could have a value of `auto` (default), `none` (always null, probably what you want here) and `config` (which is the default when no .git info is present, i.e. it hashes the composer.json and repo config, and still provides a useful reference showing when something changed in that package, which I find more useful than no ref at all).

**Comment by Seldaek** [comment]
@sveneld I saw yes, could you update it to match what I described above perhaps with the different modes?

**Comment by sveneld** [comment]
Ok i will do it tomorrow

**Comment by sveneld** [comment]
@Seldaek done
